packages/ra-ui-materialui/src/button/CloneButton.tsx

```diff
@@ -29,10 +29,10 @@ export const CloneButton = (props: CloneButtonProps) => {
                           search: stringify({
                               source: JSON.stringify(omitId(record)),
                           }),
-                          state: { _scrollToTop: scrollToTop },
                       }
                     : pathname
             }
+            state={{ _scrollToTop: scrollToTop }}
```

Comment:
This probably means we introduced a breaking change even though passing the previous `to` to the `Button` without a `component={Link}` didn't work before. I'm not sure what we should do here. @fzaninotto @slax57 @adguernier?

Reply:
I agree that this is a breaking change and we should not do it. Instead, we must deprecate the support for the to prop in button, but keep the logic there.